File: pinyougou_manager_web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

/**

 */
@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference
    private BrandService brandService;//引入dubbo的接口服务对象

    @RequestMapping("/findAll")
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }


    @RequestMapping("/findPage")
    public PageResult findPage(Integer pageNum,Integer pageSize){
        return brandService.findPage(pageNum,pageSize);
    }

    @RequestMapping("/add")
    //angularjs 传递的数据是对象类型默认的content-type:applicaton/json;
    public Result add(@RequestBody TbBrand brand){
        try {
            brandService.add(brand);//ctr+alt +t
            return new Result(true,"新增成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"新增失败");
        }
    }

    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        TbBrand brand = brandService.findOne(id);
        return brand;
    }

    @RequestMapping("/update")
    public Result update(@RequestBody TbBrand brand){
        try {
            brandService.update(brand);
            return new Result(true,"更新成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"更新失败");
        }
    }

    @RequestMapping("/delete")
    public Result delete(Long[] ids){
        try {
            brandService.delete(ids);
            return new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return  new Result(false,"删除失败");
        }
    }

    @RequestMapping("/search")
    public PageResult findPage(@RequestBody TbBrand brand,Integer pageNum ,Integer pageSize){
        return brandService.findPage(brand,pageNum,pageSize);
    }


}

File: pinyougou_manager_web/src/main/webapp/js/controller/brandController.js
//2.定义一个controller
app.controller('brandController', function ($scope,$controller,brandService) {
    $controller('baseController',{$scope:$scope})
    //发送请求去查询数据库的数据
    $scope.findAll = function () {
        brandService.findAll().success(
            function (response) {//返回的数据就是[{},{}]
                $scope.list = response;
            }
        )
    }

    //初始值
    $scope.searchEntity = {};


    //根据当前的页码 和每页显示的行数分页查询调用
    $scope.findPage = function () {
        brandService.findPage($scope.paginationConf.currentPage, $scope.paginationConf.itemsPerPage).success(
            function (response) {//response 是PageResult
                $scope.list = response.rows;//每页显示的行
                $scope.paginationConf.totalItems = response.total;//总记录数
            }
        );
    }
    //添加品牌
    $scope.save = function () {
        //如果entity中没有ID的值说明是新增
        var serviceObject = brandService.add($scope.entity);
        // var methodName="add";
        if ($scope.entity.id != null) {
            //如果entity中有id的值说明是更新
            serviceObject = brandService.update($scope.entity)
        }
        serviceObject.success(
            function (response) {//result
                if (response.success) {
                    //新增成功 刷新列表
                    $scope.reloadList();
                } else {
                    alert(response.message);
                }

            }
        )


    }

    $scope.findOne = function (id) {
        brandService.findOne(id).success(
            function (response) {//tbbrand
                $scope.entity = response;
            }
        )
    }


    //删除选中的品牌
    $scope.dele = function () {
        brandService.dele($scope.selectIds).success(
            function (response) {//返回的是result
                if (response.success) {
                    $scope.reloadList();//刷新列表
                } else {
                    alert(response.message);
                }
            }
        )
    }

    $scope.search = function (pageNum, pageSize) {
        brandService.search(pageNum, pageSize, $scope.searchEntity).success(
            function (response) {//pageResult
                $scope.list = response.rows;//每页显示的行
                $scope.paginationConf.totalItems = response.total;//总记录数
            }
        )
    }
})